beacon-chain/cache/depositsnapshot/leaf_node.go

```diff
@@ -0,0 +1,36 @@
+package depositsnapshot
+
+import "github.com/pkg/errors"
+
+// Leaf represents a leaf node holding a deposit and satisfies the MerkleTreeNode interface.
+type Leaf struct {
+	hash [32]byte
+}
+
+// GetRoot returns the root of the Merkle tree.
+func (l *Leaf) GetRoot() [32]byte {
+	return l.hash
+}
+
+// IsFull returns whether there is space left for deposits.
+func (l *Leaf) IsFull() bool {
+	return true
+}
+
+// Finalize marks deposits of the Merkle tree as finalized.
+func (l *Leaf) Finalize(deposits uint64, depth uint64) (MerkleTreeNode, error) {
+	return &Finalized{
+		deposits: 1,
+		hash:     l.hash,
+	}, nil
+}
+
+// GetFinalized returns a list of hashes of all the finalized nodes and the number of deposits.
+func (l *Leaf) GetFinalized(result [][32]byte) ([][32]byte, uint64) {
+	return result, 0
```

Comment:
This isn't implemented right? Consider a TODO or `panic("not implemented")`

Reply:
I cross referenced the EIP spec and it appears to be correct.
```
def get_finalized(self, result: List[Hash32]) -> uint:
        return 0

Reply:
Could you explain why the API for our implementation is different than the spec? Their spec only returns the deposit count

Reply:
In the python implementation the `get_finalized` from the `class Finalized` is appending a hash to the list of hashes `result` via pointers.
I decided to return the result instead of using pointers as I was worried it could be unclear and error prone. Which means each instance of `GetFinalized` must also return the list.
This can be changed to use pointers instead.